### DISCUSSION TITLE
Possible interface developments ("diff") that could help users choose between models / prompts

### DISCUSSION THREAD

**Discussion by nickynicolson** [Gene-Weaver/VoucherVision#38] [Ideas and Feature Requests]
I wonder if a "diff" style interface might help users explore what models / prompts are available and decide which to use. These show a side by side comparison of two outputs and can be added with a javascript library.
see e.g. https://quick-diff.com/ and http://incaseofstairs.com/jsdiff/

![image](https://github.com/user-attachments/assets/4ed1127e-7268-407f-8f73-1d04d56cee5b)

**Comment by Gene-Weaver** [comment]
Great idea. I was going to need something like this down the road for our human preference testing and this makes things simple. I  just added some tests, was able to get quick-diff to work inside Streamlit too. I'll try to integrate it more soon, but it definitely works. 

If you want to try the tests, you need npm and yarn installed globally. Then the `run_diff_test.py` should take care of starting the server and linking it to the Streamlit iframe.

![quick-diff](https://github.com/user-attachments/assets/2d0f64a3-fbdf-4b8d-91ad-a0497b1e2950)

**Reply by Gene-Weaver** [reply]
I forked the quick-diff repo, added my version as a submodule, so you'll need to `git pull` and `git submodule update --init --recursive` to include the new quick-diff repo
